lib/core-integration/src/Test/Integration/Scenario/API/Shelley/StakePools.hs

```diff
@@ -593,6 +596,23 @@ spec = do
                             (.> Quantity (unsafeMkPercentage 0))
                     ]
 
+        it "pools have the correct retirement information" $ \ctx -> do
+            eventually "pools have the correct retirement information" $ do
+                r <- listPools ctx arbitraryStake
+                expectResponseCode HTTP.status200 r
+                let retirementEpochIs n = \case
+                        Nothing -> False
+                        Just ei -> epochNumber ei == ApiT (EpochNo n)
+                verify r
+                    [ expectListSize 3
+                    , expectListField 0 #retirement
+                        (`shouldBe` Nothing)
+                    , expectListField 1 #retirement
+                        (`shouldSatisfy` retirementEpochIs 10)
```

Comment:
This test passed when I run it alone... but it was not passing when I was running it with a test suite. 

I started to monitor the cluster while the tests were running and my observation is that this pool that is scheduled to retire in epoch 10 disappears from the `list stake pools` at the beginning of epoch 14... :thinking:  so, we're starting with 3 pools being registered in "setup" but since epoch 14 wallet actually shows only 2 pools!

Reply:
@piotr-iohk wrote:

> This test passed when I run it alone... but it was not passing when I was running it with a test suite.
> 
> I started to monitor the cluster while the tests were running and my observation is that this pool that is scheduled to retire in epoch 10 disappears from the `list stake pools` at the beginning of epoch 14... thinking so, we're starting with 3 pools being registered in "setup" but since epoch 14 wallet actually shows only 2 pools!

I'm a little surprised that it just disappears from the list, as we don't (yet) actively perform any garbage collection of pools after they retire. If I were to take a guess, the cause would be related to the we are merging data in order to produce the list of stake pools. I'll put this next on my list for immediate investigation.

For now, I've uploaded a patch (e80b7f5b46bb8d6e94147b6f26c2c86925d092c8) which prevents any stake pools from prematurely retiring during a run of the integration test suite, which should fix this PR.

Later on, in a further PR, I'll add more integration tests to check the behaviour of pools after they retire.